test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenHandler.ValidateTokenAsyncTests.Audience.Extensibility.cs

```diff
@@ -0,0 +1,241 @@
+        public async Task ValidateTokenAsync_Audience_Extensibility(ValidateTokenAsyncAudienceExtensibilityTheoryData theoryData)
+        {
+            var context = TestUtilities.WriteHeader($"{this}.{nameof(ValidateTokenAsync_Audience_Extensibility)}", theoryData);
+
+            string jwtString = CreateTokenWithAudience(theoryData.Audience);
+            var handler = new JsonWebTokenHandler();
+
+            ValidationResult<ValidatedToken> validationResult;
+
+            if (theoryData.ThrownException is null)
+            {
+                validationResult = await handler.ValidateTokenAsync(
+                    jwtString, theoryData.ValidationParameters!, theoryData.CallContext, CancellationToken.None);
+            }
+            else
+            {
+                // The exception is thrown by the delegate, so we catch it here.
+                // Outside of testing, this could be a catch block in the calling code.
+                var exception = await Assert.ThrowsAsync<CustomInvalidAudienceException>(async () =>
+                {
+                    validationResult = await handler.ValidateTokenAsync(
+                        jwtString, theoryData.ValidationParameters!, theoryData.CallContext, CancellationToken.None);
+                });
+
+                theoryData.ThrownException.ProcessException(exception, context);
+                return;
+            }
+
+            if (validationResult.IsSuccess != theoryData.ExpectedIsValid)
+                context.AddDiff($"validationResult.IsSuccess != theoryData.ExpectedIsValid");
+
+            if (validationResult.IsSuccess)
+            {
+                theoryData.ExpectedException.ProcessNoException(context);
+
+                IdentityComparer.AreStringsEqual(validationResult.UnwrapResult().ValidatedAudience, theoryData.Audience, context);
+            }
+            else
+            {
+                theoryData.ExpectedException.ProcessException(validationResult.UnwrapError().GetException(), context);
+
+                if (validationResult.UnwrapError().GetException() is SecurityTokenInvalidAudienceException audienceException)
+                {
+                    if (theoryData.ExpectedInvalidAudience is not null)
+                        IdentityComparer.AreStringsEqual(audienceException.InvalidAudience, theoryData.ExpectedInvalidAudience, context);
+                }
+
+                TestUtilities.AssertFailIfErrors(context);
+            }
+        }
+
+        public static TheoryData<ValidateTokenAsyncAudienceExtensibilityTheoryData> ValidateTokenAsync_Audience_ExtensibilityTestCases
+        {
+            get
+            {
+                return new TheoryData<ValidateTokenAsyncAudienceExtensibilityTheoryData>
+                {
+                    new ValidateTokenAsyncAudienceExtensibilityTheoryData("DefaultDelegate_Valid_AudiencesMatch")
```

Comment:
nit: would you be open to use the theoryData.Add() model?

Reply:
Updated.